mava/components/jax/training/step.py

```diff
@@ -471,3 +472,314 @@ def required_components() -> List[Type[Callback]]:
             mava.components.jax.training.model_updating.MinibatchUpdate,
             mava.components.jax.building.adders.ParallelSequenceAdder,
         ]
+
+
+class MAPGWithTrustRegionStepNorm(Step):
```

Comment:
We would probably need to have a chat about this in Mava standup. The issue is that the entire training step is copied, which is the same as we did in the past with shared networks. When we add new features to the trainer step function we now need to implement the features in both step functions. Hopefully, we can find a slightly more straightforward way to do this without needing to copy the step code.

Reply:
Yes you are right. We should chat about it. I also also thought about it. We need to avoid having to maintain to parallel codes.

Reply:
Great @ulricharmel :raised_hands: Thanks for the effort.